File: solutions/zig/resistor-color-duo/1/resistor_color_duo.zig
const std = @import("std");

pub const ColorBand = enum { black, brown, red, orange, yellow, green, blue, violet, grey, white };

pub fn colorCode(colors: [2]ColorBand) usize {
    const a: usize = @intFromEnum(colors[0]);
    const b: usize = @intFromEnum(colors[1]);
    return a * 10 + b;
}

File: solutions/clojure/resistor-color-duo/1/src/resistor_color_duo.clj
(ns resistor-color-duo)

(def color-codes
  (zipmap ["black" "brown" "red" "orange" "yellow"
           "green" "blue" "violet" "grey" "white"]
          (range)))

(defn resistor-value [colors]
  (+ (* 10 (color-codes (first colors)))
     (color-codes (second colors))))
